src/Gql.js

```diff
@@ -0,0 +1,266 @@
+import _ from 'lodash';
+import {parse, validate, execute} from 'graphql';
+import {Connect, Str} from 'sm-utils';
+
+import {formatError} from './errors';
+import {makeSchemaFromConfig} from './makeSchemaFrom';
+
+const ONE_DAY = 24 * 3600 * 1000;
+const ENUM_PREFIX = '__ENUM__::';
+// const VAR_PREFIX = '__VAR__::';
+
+// const NO_QUOTES_REGEX = new RegExp(`^(${ENUM_PREFIX}|${VAR_PREFIX})([A-Za-z_]+)$`);
+const ENUM_REGEX = new RegExp(`^${ENUM_PREFIX}([A-Za-z_]+)$`);
+
+/**
+ * we are not using the inbuilt graphql function because it validates
+ * the graphql, which is an expensive operation
+ * return graphql(schema, query, rootValue, context, variables);
+ * taken from:
+ * @see https://github.com/graphql/graphql-js/blob/master/src/graphql.js
+ */
+function graphql({
+	schema, query, context, variables, rootValue = null, validateGraphql = false,
+}) {
+	// parse
+	let document;
+	try {
+		document = parse(query);
+	}
+	catch (syntaxError) {
+		return Promise.resolve({errors: [syntaxError]});
+	}
+
+	if (validateGraphql) {
+		const validationErrors = validate(schema, document);
+		if (validationErrors.length > 0) {
+			return Promise.resolve({errors: validationErrors});
+		}
+	}
+
+	return execute(
+		schema,
+		document,
+		rootValue,
+		context,
+		variables,
+	);
+}
+
+function convertObjToGqlArg(obj) {
+	const gqlArg = [];
+	_.forEach(obj, (value, key) => {
+		// eslint-disable-next-line no-use-before-define
+		gqlArg.push(`${key}: ${convertToGqlArg(value)}`);
+	});
+	return `${gqlArg.join(', ')}`;
+}
+
+function convertStrToGqlArg(str) {
+	const [, val] = str.match(ENUM_REGEX) || [];
+	if (val) return val;
+	// const [, prefix, val] = str.match(NO_QUOTES_REGEX) || [];
+	// if (prefix && val) return prefix === VAR_PREFIX ? '$' + val : val;
+
+	return JSON.stringify(str);
+}
+
+function convertToGqlArg(value) {
+	if (value == null) return null;
+
+	if (typeof value === 'number') return String(value);
+	if (typeof value === 'string') return convertStrToGqlArg(value);
+	if (_.isPlainObject(value)) return `{${convertObjToGqlArg(value)}}`;
+
+	return JSON.stringify(value);
+}
+
+class ApiError extends Error {}
```

Comment:
These could also be exported. Would be helpful while logging to check if an error is instance of any of these